rotkehlchen/externalapis/opensea.py

```diff
@@ -85,6 +88,44 @@ def __init__(self, database: 'DBHandler', msg_aggregator: MessagesAggregator) ->
         # https://twitter.com/LefterisJP/status/1483017589869711364
         self.session.headers.update({'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0'})  # noqa: E501
         self.collections: Dict[str, Collection] = {}
+        self.backup_key: Optional[str] = None
+
+    def maybe_get_backup_key(self) -> Optional[str]:
+        """This will attempt to fetch the backup key from our server if not already fetched"""
+        if self.backup_key is not None:
+            return self.backup_key
+
+        try:
+            response = requests.get('https://rotki.com/api/1/credentials', json={'name': 'opensea'})  # noqa: E501
+        except requests.exceptions.RequestException as e:
+            log.error(f'Could not connect to rotki server for fetching backup opensea key due to {str(e)}')  # noqa: E501
+            return None
+
+        try:
+            json_ret = json.loads(response.text)
+        except JSONDecodeError as e:
+            log.error(f'Could not decode rotki server response for opensea key: {response.text} due to {str(e)}')  # noqa: E501
+            return None
+
+        if response.status_code != 200:
```

Comment:
Shouldn't this happen before trying to decode the json response? If something goes wrong you will see the error related to the json response but not that it was caused by a non 200

Reply:
Not necessarily as our server should always return json. Returns 4XX for errors and contains an "error" key in the dict.